lib/ts/fetch.ts

```diff
@@ -298,6 +298,11 @@ export default class AuthHttpRequest {
                     }
                     logDebugMessage("doRequest: Retrying original request");
                 } else {
+                    if (response.status === AuthHttpRequest.config.invalidClaimStatusCode) {
```

Comment:
Are you sure that you don't want to check for this in the refreshing session part as well? What if the user overrides that on the backend to do claim validation and return this type of error?

Reply:
oh right
[22:04](https://supertokens.slack.com/archives/D02B887A2MQ/p1661358843883719)
I forgot this this morning
[22:04](https://supertokens.slack.com/archives/D02B887A2MQ/p1661358890232979)
umm yeah, so I'm sure. a claim validation error coming from the refresh endpoint doesn't really make sense
[22:05](https://supertokens.slack.com/archives/D02B887A2MQ/p1661358920280709)
since you'd have no way of refreshing the access token payload
[22:06](https://supertokens.slack.com/archives/D02B887A2MQ/p1661358964338409)
and since there is no way of updating the token but it fails validation the only sensible way to handle this is to remove the session... so it's essentially a 401.